File: qc-domain/src/main/java/com/eager/questioncloud/domain/verification/EmailVerificationJpaRepository.java
package com.eager.questioncloud.domain.verification;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EmailVerificationJpaRepository extends JpaRepository<EmailVerificationEntity, String> {
    Optional<EmailVerificationEntity> findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        String token, EmailVerificationType emailVerificationType);

    Optional<EmailVerificationEntity> findByResendToken(String resendToken);
}

File: qc-user/src/main/kotlin/com/eager/questioncloud/verification/infrastructure/repository/EmailVerificationJpaRepository.kt
package com.eager.questioncloud.verification.infrastructure.repository

import com.eager.questioncloud.verification.enums.EmailVerificationType
import com.eager.questioncloud.verification.infrastructure.entity.EmailVerificationEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface EmailVerificationJpaRepository : JpaRepository<EmailVerificationEntity, String> {
    fun findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        token: String,
        emailVerificationType: EmailVerificationType
    ): Optional<EmailVerificationEntity>

    fun findByResendToken(resendToken: String): Optional<EmailVerificationEntity>
}
